File: 49-group-anagrams/group-anagrams.py
class Solution:
    def groupAnagrams(self, strs: List[str]) -> List[List[str]]:
        hm = defaultdict(list)
        for s in strs:
            sorted_s = ''.join(sorted(s))
            hm[sorted_s].append(s)
        
        
        return list(hm.values())
        # ans = []
        # for v in hm.values():
        #     ans.append(v)
        # return ans

File: 49-group-anagrams/group-anagrams.cpp
class Solution {
public:
    vector<vector<string>> groupAnagrams(vector<string>& strs) {
        int n = strs.size();
        unordered_map<string, vector<string>> hm;
        vector<vector<string>> res;

        for (const auto& str : strs) {
            string temp = str;
            sort(temp.begin(), temp.end());
            hm[temp].push_back(str);
        }

        for (auto& pair : hm) {
            res.emplace_back(std::move(pair.second));
        }
        
        return res;
    }
};